### DISCUSSION TITLE
How to implement hyper request timeout

### DISCUSSION THREAD

**Discussion by kushwahashiv** [hyperium/hyper#2449] [Q&A] [answered]
Hi All

I'm facing hyper request timeout implement/migration issue. earlier implementation is similar to below
https://github.com/hyperium/hyper/issues/1097

Now I am not sure what is the best practice to implement request timeout using futures and tokio.

Any link or help will be appreciated. I tried finding but could not find one.

Thanks in advance.

/Shiv

**Comment by davidbarsky** [comment, marked as answer]
Does something like this help?

```rust
use hyper::{Client, Uri};
use tokio::time::timeout;
use std::time::Duration;

#[tokio::main]
async fn main() -> Result<(), anyhow::Error> {
    let client = Client::new();
    let future = client.get(Uri::from_static("http://httpbin.org/ip"));
    
    match timeout(Duration::from_millis(30), future).await {
        Ok(_) => println!("Got response"),
        Err(_) => println!("timed out"),
    };
    
    Ok(())
}
```

**Reply by rafrafek** [reply]
What happens when you need to download large file which takes more time than the timeout?

**Reply by seanmonstar** [reply]
You could wrap it in a [ResponseBodyTimeout](https://docs.rs/tower-http/latest/tower_http/timeout/index.html) layer, if you wanted to make the timeout per-chunk.

**Comment by tesaguri** [comment]
[hjr3/hyper-timeout](https://github.com/hjr3/hyper-timeout) may be what you want.
